prompt2model/param_selector/search_with_optuna.py

```diff
@@ -187,48 +180,13 @@ def select_from_hyperparameters(
     def _build_hp_space(
         self, hyperparameter_space: Optional[dict[str, Any]] = None
     ) -> dict[str, Any]:
-        return (
-            {
-                "min_weight_decay": hyperparameter_space.get(
-                    "min_weight_decay",
-                    DEFAULT_HYPERPARAMETERS_SPACE.get("min_weight_decay"),
-                ),
-                "max_weight_decay": hyperparameter_space.get(
-                    "max_weight_decay",
-                    DEFAULT_HYPERPARAMETERS_SPACE.get("max_weight_decay"),
-                ),
-                "min_learning_rate": hyperparameter_space.get(
-                    "min_learning_rate",
-                    DEFAULT_HYPERPARAMETERS_SPACE.get("min_learning_rate"),
-                ),
-                "max_learning_rate": hyperparameter_space.get(
-                    "max_learning_rate",
-                    DEFAULT_HYPERPARAMETERS_SPACE.get("max_learning_rate"),
-                ),
-                "min_num_train_epochs": hyperparameter_space.get(
-                    "min_num_train_epochs",
-                    DEFAULT_HYPERPARAMETERS_SPACE.get("min_num_train_epochs"),
-                ),
-                "max_num_train_epochs": hyperparameter_space.get(
-                    "max_num_train_epochs",
-                    DEFAULT_HYPERPARAMETERS_SPACE.get("max_num_train_epochs"),
-                ),
-                "save_strategy": hyperparameter_space.get(
-                    "save_strategy",
-                    DEFAULT_HYPERPARAMETERS_SPACE.get("save_strategy"),
-                ),
-                "evaluation_strategy": hyperparameter_space.get(
-                    "evaluation_strategy",
-                    DEFAULT_HYPERPARAMETERS_SPACE.get("evaluation_strategy"),
-                ),
-                "per_device_train_batch_size": hyperparameter_space.get(
-                    "per_device_train_batch_size",
-                    DEFAULT_HYPERPARAMETERS_SPACE.get("per_device_train_batch_size"),
-                ),
-            }
-            if hyperparameter_space is not None
-            else DEFAULT_HYPERPARAMETERS_SPACE
-        )
+        if hyperparameter_space is None:
+            return DEFAULT_HYPERPARAMETERS_SPACE
+
+        hp_space = {}
+        for key, default_value in DEFAULT_HYPERPARAMETERS_SPACE.items():
+            hp_space[key] = hyperparameter_space.get(key, default_value)
```

Comment:
What about the situation where the user provides a key in `hyperparameter_space` that is not found in  `DEFAULT_HYPERPARAMETERS_SPACE` (e.g. `max_grad_norm`)?

In the current implementation, this parameter will be ignored. I think there's two ways we can handle this case:
1) Add this hyperparameter to the `hp_space` dictionary anyways (I think this makes the most sense)
2) Log a warning that an unexpected hyperparameter has been provided, then ignore the parameter. This is probably the safest option but will require users to modify the default hyperparameter object if they want to pass in any new hyperparameter.

Reply:
I mean, I am iterating over `DEFAULT_HYPERPARAMETERS_SPACE` and in order to use new key in `hyperparameter_space`, the user has to change the `DEFAULT_HYPERPARAMETERS_SPACE`. So, is't that way the edge case is automatically handled?

Reply:
Yes, but this new key in `hyperparameter_space` will be silently ignored in the current implementation. That may be confusing for a user, who gets no indication that the supplied hyperparameter would be used

Reply:
Ahh got it, so for any new key the user puts, it will just log a warning right?

Reply:
Yes, I can add this.